api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirTaskDao.java

```diff
@@ -18,12 +18,14 @@
 import ca.uhn.fhir.rest.param.TokenOrListParam;
 import org.hl7.fhir.r4.model.DomainResource;
 import org.openmrs.module.fhir2.FhirTask;
+import org.openmrs.scheduler.Task;
+import org.springframework.beans.factory.annotation.Autowired;
 
-public interface FhirTaskDao {
+public interface FhirTaskDao extends FhirDao<Task> {
```

Comment:
FhirDao<?> here an OpenMRS resource should be use and Openmrs doesn't have a resource called `Task`. I think here `Order` should be used instead. And similar changes should be made in other classes as well.
@ibacher Please correct me if I am wrong.

Reply:
We do have a resource named Task. It's defined by this module. There's even another OpenMRS class called Task because, well, why not?

Reply:
Oh, wait, it should be `FhirTask`! I forgot!